packages/components/src/components/pin-code/pin-code-styles.ts

```diff
@@ -0,0 +1,107 @@
+import type { FormState } from '../../utils/form/form-state';
+import type { BreakpointCustomizable, Theme } from '../../types';
+import type { PinCodeType } from './pin-code-utils';
+import type { Styles } from 'jss';
+import { getCss, mergeDeep } from '../../utils';
+import { getLabelStyles } from '../../styles/form-styles';
+import { getFunctionalComponentRequiredStyles } from '../common/required/required-styles';
+import { getFunctionalComponentStateMessageStyles } from '../common/state-message/state-message-styles';
+import { isTypeNumber } from './pin-code-utils';
+import { getThemedFormStateColors } from '../../styles/form-state-color-styles';
+import {
+  addImportantToEachRule,
+  getThemedColors,
+  getTransition,
+  hostHiddenStyles,
+  hoverMediaQuery,
+} from '../../styles';
+import {
+  borderRadiusSmall,
+  borderWidthBase,
+  fontLineHeight,
+  spacingStaticSmall,
+  textSmallStyle,
+} from '@porsche-design-system/utilities-v2';
+
+export const getComponentCss = (
+  type: PinCodeType,
+  hideLabel: BreakpointCustomizable<boolean>,
+  state: FormState,
+  isDisabled: boolean,
+  theme: Theme
+): string => {
+  const { contrastMediumColor, disabledColor, primaryColor } = getThemedColors(theme);
+  const { formStateColor, formStateHoverColor } = getThemedFormStateColors(theme, state);
+  const inputSize = `calc(${fontLineHeight} + 10px + ${borderWidthBase} * 2 + ${spacingStaticSmall} * 2)`; // we need 10px additionally so input height becomes 54px
+  return getCss({
+    '@global': {
+      ':host': {
+        display: 'block',
+        ...addImportantToEachRule({
+          ...hostHiddenStyles,
+        }),
+      },
+      ...addImportantToEachRule({
+      input: {
```

Comment:
Can't we use the `getBaseChildStyles()` util that produces consistent styles for all other elements that look like text fields?